android/app/src/main/java/updated/mysterium/vpn/ui/manual/connect/filter/FilterActivity.kt

```diff
@@ -0,0 +1,178 @@
+package updated.mysterium.vpn.ui.manual.connect.filter
+
+import android.os.Bundle
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.LinearLayoutManager
+import network.mysterium.vpn.R
+import network.mysterium.vpn.databinding.FragmentNodeListBinding
+import network.mysterium.vpn.databinding.ToolbarBaseConnectBinding
+import org.koin.android.ext.android.inject
+import updated.mysterium.vpn.model.filter.NodePrice
+import updated.mysterium.vpn.model.filter.NodeQuality
+import updated.mysterium.vpn.model.filter.NodeType
+import updated.mysterium.vpn.model.manual.connect.CountryNodesModel
+import updated.mysterium.vpn.model.manual.connect.ProposalModel
+import updated.mysterium.vpn.ui.manual.connect.BaseConnectActivity
+
+class FilterActivity : BaseConnectActivity() {
+
+    companion object {
+
+        var countryNodesModel: CountryNodesModel? = null
+    }
+
+    private lateinit var binding: FragmentNodeListBinding
+    private val nodeListAdapter = FilterAdapter()
+    private val viewModel: FilterViewModel by inject()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = FragmentNodeListBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        initProposalListRecycler()
+        bindsActions()
+        subscribeViewModel()
+        getNodesList()
+    }
+
+    override fun reUseToolbar(toolbarBinding: ToolbarBaseConnectBinding) {
+        changeRightIcon(R.drawable.icon_search)
+        if (toolbarBinding.root.parent != null) {
+            (toolbarBinding.root.parent as ViewGroup).removeView(toolbarBinding.root)
+        }
+        binding.manualConnectToolbar.addView(toolbarBinding.root)
```

Comment:
Not sure I understand why we need this. Remove view and that add add another view. We need just show/hide buttons and that's all.

Reply:
Fix